Taverns vertical slice deployment workflow

techratTV opened the pull request:
## PR Checklist
<!-- Insert "x" inside square brackets to check item. -->

Please check if your PR fulfills the following requirements:

- [ ] All new classes have class-level documentation comments, if there are any at all
- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
<!-- Please check the one that applies to this PR. -->

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:


## What is the current behaviour?
https://github.com/Bannerlord-Coop-Team/BannerlordCoop/issues/318
No working Build files were present


## What is the new behaviour?
Added a new build file with automatic build on push to master, or a manual build from the action with a workflow_dispatch.


## Other information
Currently the build completes with no errors, however it does not put the build anywhere and no tests are ran. Will need some one to work with me or atleast inform me of where you wish these to be offloaded and how to run your test suite.

techratTV committed:
Renaming old workflow, and setting it to run on workflow_dispatch

Github action to run on windows-latest, will:
setup-msbuild
restore the nuget packages
and then attempt to build with high verbrosity

techratTV committed:
Update build_test_server.yml

techratTV committed:
Added back in the auto deploys on dispatch

Added in additional notes

garrettluskey commented on .github/workflows/build_test_server.yml:
```diff
@@ -0,0 +1,34 @@
+
+      # Run Nuget restore to ensure we have all the packages for the test server
+    - name: Nuget Restore
+      run: nuget restore source\TestServer\TestServer.csproj
```
TestServer no longer exists, the name is now IntroServer.csproj

garrettluskey commented on .github/workflows/build_test_server.yml:
```diff
@@ -0,0 +1,34 @@
+
+      # Build the test server with high Verboristy
+    - name: build Test Server
+      run: msbuild source\TestServer\TestServer.csproj -verbosity:diag
```
TestServer no longer exists, the name is now IntroServer.csproj

garrettluskey commented:
Resolved #318

techratTV committed:
Merge pull request #1 from techratTV/taverns-vertical-slice

Taverns vertical slice

techratTV committed:
Updating Build to load IntroServer Instead of TestServer

techratTV committed:
Update build_test_server.yml

Im a dummy, Forgot to change the full path, not just the file name.

garrettluskey commented on .github/workflows/build_test_server.yml:
```diff
@@ -0,0 +1,34 @@
+
+  # These were left in from the original attempt, Which should allow for automated building on pushes/pr's
+  push:
+    branches: [ master ]
```
This should be taverns-vertical-slice only

garrettluskey replied:
Only on PR is ok

garrettluskey commented on .github/workflows/build_test_server.yml:
```diff
@@ -0,0 +1,34 @@
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master, release ]
```
This should be taverns-vertical-slice only

techratTV committed:
Updating name and branch for workflow

Changed the name of the work flow and set it to run on push to taverns-vertical-slice
